subsys/bluetooth/host/ecc.c

```diff
@@ -103,6 +103,21 @@ int bt_pub_key_gen(struct bt_pub_key_cb *new_cb)
 	return 0;
 }
 
+void bt_pub_key_hci_disrupted(void)
```

Comment:
Let's use something else than "disrupted" here. I'd propose "hci_reset" since that's what it's used for.

Reply:
Can't this also be disrupted if there is a disconnect between the key being requested, and the key being received?

Reply:
Naming it "HCI reset" suggest that it's called in for every HCI reset message, including the one in bt_enable.

Reply:
> Can't this also be disrupted if there is an ACL disconnect between the key being requested, and the key being received?

No, this is not related at to connections. The key pair creation and dhkey calculation functions are done independent of any connections.



> Naming it "HCI reset" suggest that it's called in for every HCI reset message, including the one that's part of bt_enable.

Sure, if you want 100% correctness we can call the API on bt_enable() as well, I'd be fine with it. Note that we're talking about an internal API however, not something that would be visible to general users of the public Bluetooth API. If you have a better name than "hci_reset" for what happens when we reset HCI, but not when we reset HCI for the first time, I'm open to suggestions :)

Reply:
I think disrupted is better personally.

Reply:
> If you have a better name than "hci_reset" for what happens when we reset HCI, but not when we reset HCI for the first time, I'm open to suggestions :)

"Disrupted" or something similar is good I think. Maybe "HCI link lost"? "HCI uninitialized"? "HCI down"? (Inspired by ifconfig eth0 up/down.).

There may be a long time between the "link is lost", and HCI reset. The relevant event is when the HCI state machine re-enters the uninitialized / start state. That can later transition to the "enabling" state, triggering the actual HCI reset message.